libsolidity/formal/BMC.cpp

```diff
@@ -498,8 +498,39 @@ pair<smtutil::Expression, smtutil::Expression> BMC::arithmeticOperation(
 
 	auto values = SMTEncoder::arithmeticOperation(_op, _left, _right, _commonType, _expression);
 
+	IntegerType const* intType = nullptr;
+	if (auto const* type = dynamic_cast<IntegerType const*>(_commonType))
+		intType = type;
+	else
+		intType = TypeProvider::uint256();
```

Comment:
Since there's other minor coding style issues anyways, we could simplify this to

```suggestion
	auto intType = dynamic_cast<IntegerType const*>(_commonType);
	if (!intType)
			intType = TypeProvider::uint256();
```